File: WEEK 1/DAY 7 - Queues Basics/SOLUTIONS/03_Hard_Problem_Solution/Solution.py
class Solution(object):
    def simplifyPath(self, path):
        """
        :type path: str
        :rtype: str
        """
        stack = []
        parts = path.split("/")

        for part in parts:
            if part == "..":
                if stack:
                    stack.pop()  # Go back to the parent directory
            elif part == "." or part == "":
                continue  # Skip the current directory and empty parts
            else:
                stack.append(part)  # Add the valid directory name

        return "/" + "/".join(stack)

File: WEEK 1/DAY 7 - Queues Basics/SOLUTIONS/03_Hard_Problem_Solution/Solution.cpp
class Solution {
public:
    string simplifyPath(string path) {
        vector<string> stack;
        stringstream ss(path);
        string component;
        
        // Process each component in the path
        while (getline(ss, component, '/')) {
            if (component == "" || component == ".") {
                // Skip empty or current directory '.'
                continue;
            } else if (component == "..") {
                // Go up to parent directory if stack is not empty
                if (!stack.empty()) {
                    stack.pop_back();
                }
            } else {
                // Valid directory or file name, push to stack
                stack.push_back(component);
            }
        }

        // Build the simplified path
        string result = "/";
        for (int i = 0; i < stack.size(); i++) {
            if (i > 0) result += "/";
            result += stack[i];
        }
        return result;
    }
};
